src/mqtt_client.rs

```diff
@@ -383,21 +346,36 @@ where
             _ => Err(Error::Unsupported),
         }
     }
+}
 
-    fn connect_socket(&mut self) -> Result<(), Error<N::Error>> {
-        let mut socket_ref = self.socket.borrow_mut();
-        let socket = socket_ref.take().unwrap();
-
-        // Connect to the broker's TCP port with a new socket.
-        // TODO: Limit the time between connect attempts to prevent network spam.
-        let socket = self
-            .network_stack
-            .connect(socket, SocketAddr::new(self.session_state.broker, 1883))?;
+impl<N: TcpClientStack, const T: usize> Minimq<N, T> {
+    /// Construct a new MQTT client.
+    ///
+    /// # Args
+    /// * `broker` - The IP address of the broker to connect to.
+    /// * `client_id` The client ID to use for communicating with the broker. If empty, rely on the
+    ///   broker to automatically assign a client ID.
+    /// * `network_stack` - The network stack to use for communication.
+    ///
+    /// # Returns
+    /// A `Minimq` interface that can be used for publishing messages and subscribing to topics.
```

Comment:
Update comments

Reply:
They are still mostly accurate. I've updated a few parts though - let me know what else you're referring to if I missed something